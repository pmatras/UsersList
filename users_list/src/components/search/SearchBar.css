.search-bar {
  border-style: solid;
  border-width: 3px;
  border-color: gray;
}
