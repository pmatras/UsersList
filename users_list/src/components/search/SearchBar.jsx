import React from 'react';
import PropTypes from 'prop-types';

import './SearchBar.css';

const SearchBar = ({ value, onQueryChange, placeholder = 'Search...' }) => {
  return (
    <div>
      <input
        type='search'
        className='search-bar'
        placeholder={placeholder}
        value={value}
        onChange={onQueryChange}
        data-testid='search-bar'
      />
    </div>
  );
};

SearchBar.propTypes = {
  value: PropTypes.string.isRequired,
  onQueryChange: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
};

export default SearchBar;
