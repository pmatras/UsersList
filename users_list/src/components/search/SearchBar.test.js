import React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

import SearchBar from './SearchBar';

afterAll(cleanup);

describe('SearchBar component tests', () => {
  it('renders successfully', () => {
    const { getByTestId } = render(
      <SearchBar value='' onQueryChange={() => {}} />
    );
    expect(getByTestId('search-bar')).toBeInTheDocument();
  });

  it('renders successfully with passed value prop', () => {
    const searchQuery = 'Search Query';
    const { getByTestId } = render(
      <SearchBar value={searchQuery} onQueryChange={() => {}} />
    );
    expect(getByTestId('search-bar')).toHaveValue(searchQuery);
  });

  it('renders successfully with empty value passed', () => {
    const { getByTestId } = render(
      <SearchBar value='' onQueryChange={() => {}} />
    );
    expect(getByTestId('search-bar')).toHaveValue('');
  });

  it('renders successfully with default placeholder', () => {
    const { getByPlaceholderText } = render(
      <SearchBar value='' onQueryChange={() => {}} />
    );
    expect(getByPlaceholderText('Search...')).toBeInTheDocument();
  });

  it('renders successfully with custom placeholder', () => {
    const placeholder = 'Type to search...';
    const { getByPlaceholderText } = render(
      <SearchBar value='' onQueryChange={() => {}} placeholder={placeholder} />
    );
    expect(getByPlaceholderText(placeholder)).toBeInTheDocument();
  });

  it('properly fires onQueryChange function', () => {
    const onQueryChange = jest.fn();
    const { getByTestId } = render(
      <SearchBar value='' onQueryChange={onQueryChange} />
    );
    const input = 'Typing...';
    userEvent.type(getByTestId('search-bar'), input);
    expect(onQueryChange).toBeCalledTimes(input.length);
  });

  it('matches snapshot with empty value', () => {
    const tree = renderer
      .create(<SearchBar value='' onQueryChange={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot with provided value', () => {
    const tree = renderer
      .create(<SearchBar value='Example query' onQueryChange={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
