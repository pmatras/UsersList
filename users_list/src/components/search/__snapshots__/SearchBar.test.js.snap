// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component tests matches snapshot with empty value 1`] = `
<div>
  <input
    className="search-bar"
    data-testid="search-bar"
    onChange={[Function]}
    placeholder="Search..."
    type="search"
    value=""
  />
</div>
`;

exports[`SearchBar component tests matches snapshot with provided value 1`] = `
<div>
  <input
    className="search-bar"
    data-testid="search-bar"
    onChange={[Function]}
    placeholder="Search..."
    type="search"
    value="Example query"
  />
</div>
`;
